SodiumTests/SodiumTests.swift

```diff
@@ -284,4 +284,47 @@ class SodiumTests: XCTestCase {
         let verify2 = sodium.auth.verify(message: message, secretKey: key2, tag: tag)
         XCTAssertFalse(verify2)
     }
+	
+	func testKeyDerivationInputKeyTooShort() throws {
+		let sodium = Sodium()!
+		let seed = sodium.utils.hex2bin("00 11 22 33 44 55 66 77 88 99 aa bb cc dd ee ff 00 11 22 33 44 55 66 77 88 99 aa bb cc dd ee ff", ignore: " ")!
+		let inputKey = sodium.randomBytes.deterministic(length: 15, seed: seed)!
```

Comment:
I think deterministically generating random data is not required here. See also below.